project_generator/project.py

```diff
@@ -490,7 +468,8 @@ def generate_dict_for_tool(self, tool):
             'template': toolchain_specific_settings.template or [
                 tool_settings.template for tool_settings in tool_specific_settings if tool_settings.template],
         }
-        self.validate_generated_dic(d)
+        if d['linker_file'] == None and d['output_type'] == 'exe':
```

Comment:
Yes, I intended to add more things there. But haven't got time yet to consider what should be in the dictionary before it goes to exporter.